flight/targets/sparky/fw/pios_board.c

```diff
@@ -594,263 +395,39 @@ void PIOS_Board_Init(void) {
-		break;
-	case HWSPARKY_MAINPORT_DSM:
-#if defined(PIOS_INCLUDE_DSM)
-		{
-			PIOS_Board_configure_dsm(&pios_main_dsm_hsum_cfg, &pios_main_dsm_aux_cfg, &pios_usart_com_driver,
-				MANUALCONTROLSETTINGS_CHANNELGROUPS_DSM, &hw_DSMxMode);
-		}
-#endif	/* PIOS_INCLUDE_DSM */
-		break;
-	case HWSPARKY_MAINPORT_DEBUGCONSOLE:
-#if defined(PIOS_INCLUDE_DEBUG_CONSOLE) && defined(PIOS_INCLUDE_USART) && defined(PIOS_INCLUDE_COM)
-		PIOS_Board_configure_com(&pios_main_usart_cfg, 0, PIOS_COM_DEBUGCONSOLE_TX_BUF_LEN, &pios_usart_com_driver, &pios_com_aux_id);
-#endif	/* PIOS_INCLUDE_DEBUG_CONSOLE */
-		break;
-	case HWSPARKY_MAINPORT_COMBRIDGE:
-#if defined(PIOS_INCLUDE_USART) && defined(PIOS_INCLUDE_COM)
-		PIOS_Board_configure_com(&pios_main_usart_cfg, PIOS_COM_BRIDGE_RX_BUF_LEN, PIOS_COM_BRIDGE_TX_BUF_LEN, &pios_usart_com_driver, &pios_com_bridge_id);
-#endif
-		break;
-	case HWSPARKY_MAINPORT_MAVLINKTX:
-#if defined(PIOS_INCLUDE_MAVLINK)
-		PIOS_Board_configure_com(&pios_main_usart_cfg, 0, PIOS_COM_MAVLINK_TX_BUF_LEN, &pios_usart_com_driver, &pios_com_mavlink_id);
-#endif  /* PIOS_INCLUDE_MAVLINK */
-		break;
-	case HWSPARKY_MAINPORT_MAVLINKTX_GPS_RX:
-#if defined(PIOS_INCLUDE_GPS)
-#if defined(PIOS_INCLUDE_MAVLINK)
-		PIOS_Board_configure_com(&pios_main_usart_cfg, PIOS_COM_GPS_RX_BUF_LEN, PIOS_COM_MAVLINK_TX_BUF_LEN, &pios_usart_com_driver, &pios_com_gps_id);
-		pios_com_mavlink_id = pios_com_gps_id;
-#endif  /* PIOS_INCLUDE_MAVLINK */
-#endif  /* PIOS_INCLUDE_GPS */
-		break;
-	case HWSPARKY_MAINPORT_HOTTTELEMETRY:
-#if defined(PIOS_INCLUDE_HOTT) && defined(PIOS_INCLUDE_USART) && defined(PIOS_INCLUDE_COM)
-		PIOS_Board_configure_com(&pios_main_usart_cfg, PIOS_COM_HOTT_RX_BUF_LEN, PIOS_COM_HOTT_TX_BUF_LEN, &pios_usart_com_driver, &pios_com_hott_id);
-#endif /* PIOS_INCLUDE_HOTT */
-		break;
-	case HWSPARKY_MAINPORT_FRSKYSENSORHUB:
-#if defined(PIOS_INCLUDE_FRSKY_SENSOR_HUB) && defined(PIOS_INCLUDE_USART) && defined(PIOS_INCLUDE_COM)
-		PIOS_Board_configure_com(&pios_main_usart_sport_cfg, 0, PIOS_COM_FRSKYSENSORHUB_TX_BUF_LEN, &pios_usart_com_driver, &pios_com_frsky_sensor_hub_id);
-#endif /* PIOS_INCLUDE_FRSKY_SENSOR_HUB */
-		break;
-	case HWSPARKY_MAINPORT_LIGHTTELEMETRYTX:
-#if defined(PIOS_INCLUDE_LIGHTTELEMETRY)
-	PIOS_Board_configure_com(&pios_main_usart_cfg, 0, PIOS_COM_LIGHTTELEMETRY_TX_BUF_LEN, &pios_usart_com_driver, &pios_com_lighttelemetry_id);
-#endif /* PIOS_INCLUDE_LIGHTTELEMETRY */
-		break;
-	case HWSPARKY_MAINPORT_FRSKYSPORTTELEMETRY:
-#if defined(PIOS_INCLUDE_FRSKY_SPORT_TELEMETRY)
-		PIOS_Board_configure_com(&pios_main_usart_sport_cfg, PIOS_COM_FRSKYSPORT_RX_BUF_LEN, PIOS_COM_FRSKYSPORT_TX_BUF_LEN, &pios_usart_com_driver, &pios_com_frsky_sport_id);
-#endif /* PIOS_INCLUDE_FRSKY_SPORT_TELEMETRY */
-		break;
-	}
+	/* Configure main USART port */
+	uint8_t hw_mainport;
+	HwSparkyMainPortGet(&hw_mainport);
+	PIOS_HAL_ConfigurePort(hw_mainport, &pios_main_usart_cfg,
+												 &pios_usart_com_driver, NULL, NULL, NULL,
```

Comment:
These are super deep.  2 tabs (or sometimes 1, if it is justified and looks OK) beyond the previous line's indent for a continuation is good).